File: Ruby/bubble_sort.rb
def bubble_sort(arr)
  length = arr.length - 1

  (0..length).each do |i|
    (0..length - i - 1).each do |j|
      next_i = j + 1
      arr[next_i], arr[j] = arr[j], arr[next_i] if arr[j] > arr[next_i]
    end
  end

  arr
end

File: PHP/bubbleSort.php
<?php

function bubble_sort($arr) {
    $len = count($arr)-1;
    for ($i=0; $i<$len; $i++) {
        for ($j=0; $j<$len-$i; $j++) {
            $k = $j+1;
            if ($arr[$k] < $arr[$j]) {
                // Swap elements at indices: $j, $k
                list($arr[$j], $arr[$k]) = array($arr[$k], $arr[$j]);
            }
        }
    }
    return $arr;
}

$arr = array(64,34,25,12,22,11,90);

print("<pre>Before sorting\n");
print_r($arr);
print("After sorting by using bubble sort\n");
print_r(bubble_sort($arr));

?>